synfig-studio/src/gui/CMakeLists.txt

```diff
@@ -28,7 +28,12 @@ endif()
 ## Targets
 ##
 
-add_executable(synfigstudio main.cpp)
+add_executable(synfigstudio main.cpp ${CMAKE_CURRENT_BINARY_DIR}/../../ui/resource.c)
+add_dependencies(synfigstudio build_resource)
+set_source_files_properties(
+	${CMAKE_CURRENT_BINARY_DIR}/../../ui/resource.c
```

Comment:
I'm not a CMake guy at all, so please ignore if it is a stupid question: is it the better way to point it to `resource.c`?
And should it be referred twice?

Reply:
Yes, resource.c is a generated file
At line 32 synfigstudio depends on build_resource target `add_dependencies(synfigstudio build_resource)`
So before the first arg is build second arg is built first but only after the user executes make command and not cmake

Now CMake doesn't know where resource.c is coming from because it isn't generated yet
That's why we are explicitly telling CMake to not fail when generating build files by turning on resource.c GENERATED file property as true

Reply:
There is an alternative though

We can instead remove that resource.c path from line 31 and also eliminate line 33-35 and instead insert resource.c as a header file inside canvas_properties.h file. The problem would be that IDE will always redline that include file because it doesn't exist in src dir (we're doing out-of-source build). Plus we have to move the generated resource.c file from src/synfig-studio/ui build dir to src/synfig-studio/src/gui/dialogs build dir

Using the above method will make CMake script look clean and readable because TBH the current one is ugly now but effective

Reply:
I really don't know CMake :( I trust you, I just thought it was strange :)

Maybe @ice0 could opine here